docs/src/background/algebraic_modeling_languages.md

```diff
@@ -0,0 +1,185 @@
+
+For example, here's how we would write the knapsack problem in JuMP:
+```jldoctest
+julia> function algebraic_knapsack(c, w, b)
+           n = length(c)
+           model = Model()
+           @variable(model, x[1:n] >= 0, Int)
+           @objective(model, Min, sum(c[i] * x[i] for i = 1:n))
+           @constraint(model, sum(w[i] * x[i] for i = 1:n) <= b)
+           return print(model)
+       end
+algebraic_knapsack (generic function with 1 method)
+
+julia> algebraic_knapsack([1, 2], [0.5, 0.5], 1.25)
+Min x[1] + 2 x[2]
+Subject to
+ 0.5 x[1] + 0.5 x[2] ≤ 1.25
+ x[1] ≥ 0.0
+ x[2] ≥ 0.0
+ x[1] integer
+ x[2] integer
+```
+This formulation is compact, and it closely matches the algebraic formulation of
+the model we wrote out above.
+
+Here's what the JuMP code would look like if we didn't use macros:
+```jldoctest
+julia> function nonalgebraic_knapsack(c, w, b)
+           n = length(c)
+           model = Model()
+           x = [VariableRef(model) for i = 1:n]
+           for i = 1:n
+               set_lower_bound(x[i], 0)
+               set_integer(x[i])
+               set_name(x[i], "x[$i]")
+           end
+           obj = AffExpr(0.0)
+           for i = 1:n
+               add_to_expression!(obj, c[i], x[i])
+           end
+           set_objective(model, MOI.MIN_SENSE, obj)
+           lhs = AffExpr(0.0)
+           for i = 1:n
+               add_to_expression!(lhs, w[i], x[i])
+           end
+           con = build_constraint(error, lhs, MOI.LessThan(b))
+           add_constraint(model, con)
+           return print(model)
+       end
+nonalgebraic_knapsack (generic function with 1 method)
+
+julia> nonalgebraic_knapsack([1, 2], [0.5, 0.5], 1.25)
+Min x[1] + 2 x[2]
+Subject to
+ 0.5 x[1] + 0.5 x[2] ≤ 1.25
+ x[1] ≥ 0.0
+ x[2] ≥ 0.0
+ x[1] integer
+ x[2] integer
+```
+
+Hopefully you agree that the macro version is much easier to read!
+
+## Part II: talking to solvers
+
+Another complicating factor is the diversity of standard forms expected by
+solvers. For example, other linear programming solvers require a standard form
+like:
+```math
+\begin{aligned}
+\min & c^\top x
+\text{s.t.} & l_c \le A x \le u_c \\
+            & l_x \le x \le u_x,
+\end{aligned}
+```
+
+```math
+\begin{aligned}
+\min & c^\top x
+\text{s.t.} & A x - b \in \mathcal{K}\\
+\end{aligned}
+```
+```math
+\begin{aligned}
+\min & c^\top x
+\text{s.t.} & A x = b \\
+            & x \in \mathcal{K}
+\end{aligned}
+```
+
+## What is a Solver?
+
+A solver is a software package that incorporates algorithms for finding
+solutions to one or more classes of problem.
+
+For example, GLPK is a solver for linear programming (LP) and mixed integer
+programming (MIP) problems. It incorporates algorithms such as the simplex
+method and the interior-point method.
+
+JuMP currently supports a number of open-source and commercial solvers, which
+can be viewed in the [Supported-solvers](@ref) table.
```

Comment:
Perhaps this is a good place to say that solvers are 100% responsible for the quality of the solution.

Reply:
Yip. this part is still very much WIP.